File: examples/pulsar-quickstart/src/main/java/acme/BeanUsingAnEmitter.java
package acme;

import org.eclipse.microprofile.reactive.messaging.Channel;
import org.eclipse.microprofile.reactive.messaging.Emitter;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;
import java.util.concurrent.Executors;
import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicInteger;

@ApplicationScoped
public class BeanUsingAnEmitter {

    @Inject
    @Channel("my-channel")
    Emitter<String> emitter;

    public void periodicallySendMessageToPulsar() {
        AtomicInteger counter = new AtomicInteger();
        Executors.newSingleThreadScheduledExecutor()
                .scheduleAtFixedRate(() -> {
                    emitter.send("Hello " + counter.getAndIncrement());
                },
                        1, 1, TimeUnit.SECONDS);
    }

}

File: examples/kafka-quickstart-kotlin/src/main/kotlin/acme/BeanUsingAnEmitter.kt
package acme

import org.eclipse.microprofile.reactive.messaging.Channel
import org.eclipse.microprofile.reactive.messaging.Emitter
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit
import java.util.concurrent.atomic.AtomicInteger

import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject


@ApplicationScoped
open class BeanUsingAnEmitter {

    @Inject
    @Channel("my-channel")
    private lateinit var emitter: Emitter<String>

    open fun periodicallySendMessageToKafka() {
        val counter = AtomicInteger()
        Executors.newSingleThreadScheduledExecutor()
                .scheduleAtFixedRate({ emitter.send("Hello " + counter.getAndIncrement()) },
                        1, 1, TimeUnit.SECONDS)
    }

}
